src/verbs/completion_queue.rs

```diff
@@ -54,3 +61,66 @@ impl Drop for CompletionQueue {
 pub struct CompletionQueueEx {
     pub(crate) cq_ptr: *mut ibv_cq_ex,
 }
+
+#[derive(Debug)]
+pub struct ExtendedCompletionQueue<'ctx> {
```

Comment:
I wonder what should we name the ex APIs and structs? `ExtendedXXX` or `XXXExtended`? As after doc generation, `XXX` and `XXXExtended` would be neighbor, but for `ExtendedXXX`, all extended APIs and structs would be neighbor.

It would be like this in the doc:

```
CompletionQueue
ExtendedCompletionQueue
ExtendedQueuePair
QueuePair
```
Or
```
CompletionQueue
CompletionQueueExtended
QueuePair
QueuePairExtended
```
I would prefer the second one.

Reply:
Yeah I agree with you. Making the `Extended` a suffix will make more sense.